### PR TITLE
Fix event multiplexer pattern for NATS

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- fix the event multiplexer pattern for NATS backend
- add tests to verify it

**Related issue(s)**

See also: #11719 #11989

### CODE REVIEW

**Comment by mfaizanse** [comment]
Hi,
* I have tested the PR on the cluster and the multiplexing is working. I used 3 functions and 3 subscriptions.
* I also reproduced the bug on kyma:main, and multiplexing is not working on main.
* Also, the unit test is failing without the fix, so thats good. And with new changes the unit test is passing.

File: components/eventing-controller/pkg/handlers/nats_test.go [hunk 1]
```diff
@@ -180,6 +181,107 @@ func TestSubscription(t *testing.T) {
...
+
+	// Prepare event-type cleaner
+	application := applicationtest.NewApplication(eventingtesting.ApplicationNameNotClean, nil)
+	applicationLister := fake.NewApplicationListerOrDie(context.Background(), application)
+	cleaner := eventtype.NewCleaner(eventingtesting.EventTypePrefix, applicationLister, defaultLogger)
+
+	// Create 3 subscriptions having the same sink and the same event type
+	var subs [3]*eventingv1alpha1.Subscription
+	for i:=0; i< len (subs); i++ {
+		subs[i] = eventingtesting.NewSubscription("sub" + strconv.Itoa(i), "foo", eventingtesting.WithNotCleanEventTypeFilter)
```
Comment (k15r) [inline components/eventing-controller/pkg/handlers/nats_test.go:233]: nit: `fmt` is already imported, so 
```
fmt.Sprintf("sub-%d", i)
```
```suggestion
		subs[i] = eventingtesting.NewSubscription(fmt.Sprintf("sub-%d", i), "foo", eventingtesting.WithNotCleanEventTypeFilter)
```

File: components/eventing-controller/pkg/handlers/nats.go [hunk 2]
```diff
@@ -122,7 +122,8 @@ func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner even
 		}
 
 		for i := 0; i < subsConfig.MaxInFlightMessages; i++ {
-			natsSub, subscribeErr := n.connection.QueueSubscribe(subject, subject, callback)
+			queueGroupName := createKeyPrefix(sub) + string(types.Separator) + subject
```
Comment (k15r) [inline components/eventing-controller/pkg/handlers/nats.go:125]: Could we add a comment why we introduced this prefix?

Comment (radufa) [reply components/eventing-controller/pkg/handlers/nats.go:126]: Done

File: components/eventing-controller/pkg/handlers/nats_test.go (hunk 1 above)
Comment (radufa) [reply components/eventing-controller/pkg/handlers/nats_test.go]: Done

**Review by k15r — DISMISSED** [review]

**Review by k15r — APPROVED** [review]

**Pull request merged**